ios/CustomerioReactnative.swift

```diff
@@ -120,6 +135,87 @@ class CustomerioReactnative: NSObject {
     func registerDeviceToken(token: String) -> Void {
         CustomerIO.shared.registerDeviceToken(token)
     }
+
+    /**
+     To show push notification prompt  if current authorization status is not determined
+     */
+    @objc(showPromptForPushNotifications:resolver:rejecter:)
+    func showPromptForPushNotifications(options : Dictionary<String, AnyHashable>, resolver resolve: @escaping(RCTPromiseResolveBlock),  rejecter reject: @escaping(RCTPromiseRejectBlock)) -> Void {
+        
+        // Show prompt if status is not determined
+        getPushNotificationPermissionStatus { status in
```

Comment:
What is the need to get the permission status before showing the prompt? 

I was expecting to just see [this code](https://github.com/customerio/customerio-reactnative/pull/101/files#diff-a8604755a2ab9ef911351d0719534270b59d25bb96364d6bffce4e8f8e4e9db3R174-R192) in this function. I don't think we need the rest of this code?

Reply:
I added this code so that the customer might perform action based on the promise returned. If permission has already been granted, this method will return the status in the promise that can be handled at app end. This adds better flexibility to `showPromptForPushNotifications` method.

Reply:
Ok. I can see value in doing this for RN customers to help abstract away the iOS specifics of push permissions.